### PR TITLE
add knex example

### PR DESCRIPTION
**Pull request by Noviny**
This PR introduces a copy of the `todo` example, but configured to use `knex` instead of `mongoose` - this is mostly being added to make keystone team developers have an easier time with their testing, but is also a good starting example for using `knex`

### CODE REVIEW

**Comment by Noviny** [comment]
Changelog and readme need a scrub

File: examples/todo-knex/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Demo Project: Todo List (Knex)
+
+This is the todo list - the simplest implementation of Keystone. The todo list allows you to see what the minimum requirements are for running Keystone. It has no fancy extras and only one list - easy peasy! This is a copy of our normal todo project, but configured to use our knex adapter.
+
+## Running the Project.
+
+To run this demo project, all you need to do is run `yarn` within the Keystone project root to install all required packages, then run `yarn start todo` to begin running Keystone.
```
Comment (timleslie) [inline examples/todo-knex/README.md:7]: Is `yarn start todo` the right thing here?

File: examples/todo-knex/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const { Keystone } = require('@keystonejs/keystone');
+const { KnexAdapter } = require('@keystonejs/adapter-knex');
+const { Text } = require('@keystonejs/fields');
+const { GraphQLApp } = require('@keystonejs/app-graphql');
+const { AdminUIApp } = require('@keystonejs/app-admin-ui');
+const { StaticApp } = require('@keystonejs/app-static');
+
+const keystone = new Keystone({
+  adapter: new KnexAdapter({ knexOptions: { connection: 'postgres://localhost/to-do' } }),
```
Comment (timleslie) [inline examples/todo-knex/index.js:9]: We should give some instructions on how to create this database in the README.md

**Review by timleslie — CHANGES_REQUESTED** [review]

**Comment by Noviny** [comment]
Good feedback. Updated readmes where relevant.

**Review by timleslie — APPROVED** [review]
LGTM 👍

**Pull request merged**